hack/clickhouse-backup-restore/main.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"antrea.io/theia/hack/clickhouse-backup-restore/clickhouseBackupRestore"
+)
+
+func main() {
+	// Define command-line flags for namespace and pod name.
+	fromYaml := flag.String("fromYaml", "", "YAML file for cluster update from")
+	toYaml := flag.String("toYaml", "", "YAML file for cluster update to")
+	deleteBackup := flag.Bool("deleteBackup", true, "Delete Backup after success")
+	deleteYamls := flag.Bool("deleteYamls", false, "Delete Yamls after success")
+	loglevel := flag.String("loglevel", "0", "Set loglevel 1 for debug")
+	flag.Parse()
+
+	fmt.Println("Cluster update from:", *fromYaml)
+	fmt.Println("Cluster update to:", *toYaml)
+
+	// Check if to and from yamls are provided.
+	if *toYaml == "" || *fromYaml == "" {
+		fmt.Println("Please provide both YAML files for upgrade ")
+		os.Exit(1)
+	}
+	clickhouseBackupRestore.BackupAndRestoreClickhouse(*fromYaml, *toYaml, *deleteBackup, *loglevel, *deleteYamls)
```

Comment:
I haven't looked into the details, but generally I have the following considerations

- I notice you are using a third party open source code from https://github.com/AlexAkulov/clickhouse-backup. I'm not quite sure about whether there will be a license issue as they are using an MIT License while we are using the Apache License. cc @elton-furtado for suggestions
- Can we use shell instead of golang? Otherwise users have to install golang and compile it locally only for ClickHouse upgrading/downgrading. If the logic is too complex to be implemented in shell script, maybe we can think about moving it to theia-cli or adding it as a separate binary in release.
- The script expects users to install Theia with Yaml. But it might not work well for users using Helm to do version control on Theia. Can we also add options for Helm chart upgrading?

Reply:
we will have to add MIT license in our documentation, if we ever add this support